packages/@aws-cdk/custom-resources/lib/aws-custom-resource/aws-custom-resource.ts

```diff
@@ -98,6 +98,47 @@ export interface AwsSdkCall {
   readonly outputPath?: string;
 }
 
+/**
+ * The IAM Policy that will be applied to the different calls.
+ */
+export class AwsCustomResourcePolicy {
+
+  /**
+   * Use this constant to configure access to ALL resources.
+   */
+  public static readonly ALL_RESOURCES = ['*'];
+
+  /**
+   * Explicit IAM Policy Statements.
+   *
+   * @param statements the statements to propagate to the SDK calls.
+   */
+  public static fromStatements(statements: iam.PolicyStatement[]) {
+    return new AwsCustomResourcePolicy(statements, undefined);
+  }
+
+  /**
+   * Generate IAM Policy Statements from the configured SDK calls.
+   *
+   * Each SDK call with be translated to an IAM Policy Statement in the form of: `call.service:call.action` (e.g `s3:PutObject`).
+   * The `resources` of the statement will all be the same and are configured using the `resources` param.
+   *
+   * Its best to use specific resource ARN's when possible. However, you can also use `AwsCustomResourcePolicy.ALL_RESOURCES`
+   * to allow access to all resources, for example when `onCreate` is used to create a physical resource.
+   *
+   * @param resources the resources that the calls will have access to.
+   */
+  public static fromSdkCalls(resources: string[]) {
```

Comment:
I think this should not be a positional argument but rather a struct (`fromSdkCalls({ resources: '*' })`). 

Reasons:
* It will make sure the call site is clearly read (in `fromSdkCalls('*')`  its not clear what is "*"). Normally we use positional arguments when the meaning of the first argument is clear from the context of the method name. You could do `fromSdkCallsWithResources` but that's a mouthful.
* It is more extensible (in the future maybe we can offer some more advanced tools for generating policies from sdk calls) and this won't require breaking the API.

Reply:
Not sure the extensibility argument stands because you could add optional arguments that don't break the API just like you would keys in the struct.

But I definitely agree with your clarity reasoning, it's a nice pattern.